### PR TITLE
Adds H.E.C.K. suits, spraycan_paintable component, and reworks item's attackby()

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Added the H.E.C.K. suit, a guaranteed loot frop from Bubblegum.
add: The H.E.C.K. suit is fully fire (and ash) proof, and has very good melee armor.
add: H.E.C.K. suits can also be painted with spraycans, to fully customize your experience.
add: Despite spending centuries inside a demon king, H.E.C.K. suits are most definitely safe.
/:cl:

Idea and sprites from @Mey-Ha-Zah .

hostile_suits

Closes #30997

### CODE REVIEW

**Comment by Militaires** [comment]
The mark IV hazardous environment suit has been redesigned for comfort and utility ... Oh dear.

# **HEY DOC WE DONT HAVE TIME FOR THIS!!**

**Comment by GunHog** [comment]
Miners wearing this while blasting explosive KA shots at literal giant demons. 
### WE DOOM NOW!!

File: code/modules/mining/equipment/explorer_gear.dm [hunk 1]
```diff
@@ -46,3 +46,75 @@
...
+		spraycan.use_charges(2)
+		add_atom_colour(spraycan.paint_color, FIXED_COLOUR_PRIORITY)
+		playsound(get_turf(user), 'sound/effects/spray.ogg', 5, 1, 5)
+		to_chat(user, "<span class='notice'>You sprays [spraycan] on [src], painting it.</span>")
+	else
+		return ..()
+
+/obj/item/clothing/head/helmet/space/hostile_environiment
+	name = "hostile environiment helmet"
+	desc = "A helmet with an experimental protective layer of plasma gel to protect against the hazards of Lavaland."
```
Comment (DoctoreMad) [inline code/modules/mining/equipment/explorer_gear.dm]: This lore just gets dumber and dumber. Plasma Gel?

Comment (Militaires) [reply code/modules/mining/equipment/explorer_gear.dm]: plasma-flavored lube, my favorite!

Comment (XDTM) [reply code/modules/mining/equipment/explorer_gear.dm]: that can change, just needs a better idea

File: code/modules/mining/equipment/explorer_gear.dm [hunk 2]
```diff
@@ -46,3 +46,75 @@
 /obj/item/clothing/mask/gas/explorer/folded/New()
```
Comment (PraiseRatvar) [inline code/modules/mining/equipment/explorer_gear.dm]: it's probably unrelated to the main pr but can you change this to initialize() while you're at it?

**Comment by Nabski89** [comment]
I don't suppose it could get slowdown on the station but not in low pressure like lavaland to encourage people from just making Armour?

Defenses wise it's slightly better than the syndicate hardsuit, but you know. R&D Printable.

That being said the sprites are lovely.

File: code/modules/mining/equipment/explorer_gear.dm (hunk 1 above)
Comment (DoctoreMad) [reply code/modules/mining/equipment/explorer_gear.dm]: Oh.
Let me try and see if I can come with something decent.
"Exosuit style platsteel armor"
"slime jelly encapsulation"

EDIT: I apologize

**Comment by KingPhilipIII** [comment]
Ashdrakes will be incredibly cheeseable since their attacks rely on fire.

**Comment by Hatfish** [comment]
To be fair they're already incredibly cheeseable if xenobio rushes blue slimes for cooling gel.

**Comment by XDTM** [comment]
Also they still do damage on a direct hit

**Comment by Mey-Ha-Zah** [comment]
TBH an ash drake is cheesable if you just have a fire extinguisher.

**Comment by DoctoreMad** [comment]
That's not cheese, that's how you're supposed to fight them.

**Comment by Repukan** [comment]
I'd rather these be behind the GBP machine at something like 30k points than keep powercreeping RnD with armored suits.

**Comment by Repukan** [comment]
Also mineral price is a meme because it costs nothing for RnD to print anything since everything is hand delivered to them.

**Comment by XDTM** [comment]
that's only if miners are competent, which may be a given if you always play miner, but often there's a set of miners who just die silently and never deliver

also it's miners who technically benefit from this, so technically it's not free: they mine the minerals, they get the suit

**Comment by Repukan** [comment]
Any idiot can mine so no you don't need someone competent to be completely secluded in RnD God forbid you step out for one second and get the materials yourself.

We already got a hardsuit with free flying added under the pretense of being a miner buff and I could bet you my left but I could look up the stats for how often they're made and they would concede with all the times kevinz plays since he's the only one that prints them.

Ash proofing is too good to be given to RnD as a printable. It should either be earned through extreme attrition or being able to kill a drake, and RnD in its current state is not engaging or difficult enough to warrant giving such an upgrade to scientists (who will be the mainly benefitted after all.)

**Comment by DoctoreMad** [comment]
He's saying you can easily print out armor for most of the station.

**Comment by SgtxRootbeer** [comment]
@Repukan the reasoning behind the flightsuits being useless is because they take up a backpack slot for their functionality and they cost so goddamn much, in addition to requiring you deliver your mats to a second dept (robotics) to get your speshul suit.

**Comment by FantasticFwoosh** [comment]
**Breaking News!** : Moonlighting and scientist demand harrassment rises 2000% in the face of surging product demand for hazard armor.
_________________________________________

> Holy shit. My miner slave came into my lab to bring me a crate of minerals and I literally screamed at her and hit the plate of diamonds out of her hand. She started yelling and swearing at me and I slammed the shutters on her. 

> I’m so distressed right now I don’t know what to do. I didn’t mean to do that to my only miner but I’m literally in shock from the results tonight. I feel like I’m going to explode. Why the fucking fuck is he merging this PR?

> This can’t be happening. I’m having a fucking breakdown. I don’t want to believe the world is so corrupt. I want a future to believe in. I want Armhulen to be president and fix this Moonlighting epidemic. I cannot fucking deal with this right now. It wasn’t supposed to be like this, I thought he was posting PR's well in Hippiestation ???? This is so fucked.

**Comment by KorPhaeron** [comment]
If you're desperate to add doom armour you could just bubblegum drop it in addition to his chest

They're pretty cheap and are likely going to just be mass produced by scientists for themselves if theyr'e in RD

Also cuts out some of the only legitimate work for xenobio and robotics (making fireproof people)

**Comment by Steelpoint** [comment]
We need Doom armour.

But this will become the standard issue Scientist uniform.

**Comment by ChangelingRain** [comment]
if it's specifically a bubblegum drop it shouldn't be greyscale/colorable since it'll look way better if it isn't

**Comment by XDTM** [comment]
If i fluffed in a slowdown on station but not on lavaland, would it be acceptable?

**Comment by Bawhoppen** [comment]
just don't add them to R&D goddamnit

**Comment by Steelpoint** [comment]
Honestly just make them a high level Mining item you can buy.

**Comment by XDTM** [comment]
I mean, they'd be even easier to get if they were only gated by mining points

**Comment by Repukan** [comment]
Gate it behind 60k points or more.

**Comment by Mey-Ha-Zah** [comment]
I have never achieved more than 30k mining points. But drawbacks in a high atmosphere do sound reasonable. Maybe halving its defensive stats in atmosphere could work.

**Comment by Repukan** [comment]
Drawbacks in a high atmosphere are completely null when breaking a window will give you an almost better fireproof syndie combat suit. Or just simply drain air with the machines you get in toxins.

It's so fucking easy to circumvent and Science already gets weapons, it doesn't need military grade armor too.

**Comment by DoctoreMad** [comment]
>breaking a window

Wait are you saying people will depressurize the station just so you can walk while armored or?

**Comment by Repukan** [comment]
You can very easily use the weird blue machine with a fan whose name escapes me right now to depressurize rooms.

I wouldn't put it past scientists that print guns and armor to depressurize rooms to be able to use the armor.

**Comment by Nabski89** [comment]
But that at least limits it to traitorish or last resort armor rather than everyone wants it all the time.

**Comment by Repukan** [comment]
😂

**Comment by Mey-Ha-Zah** [comment]
Alright guess just make it a drop from Bubblegum. You get one.

**Comment by MMMiracles** [comment]
if you're gonna make doom armor just go full-out with the reference and have some demon research ruin with the armor inside spooky demon rune rock

**Comment by KorPhaeron** [comment]
Or yeah, that'd be cool as well

**Comment by Mey-Ha-Zah** [comment]
Alright guess I'll design a facility

**Comment by Mey-Ha-Zah** [comment]
Maybe have the actual armor which is powerful, and a mass printable version that is only +10 defence in all directions but looks cool.

**Comment by Repukan** [comment]
RnD doesn't need fucking printable armor.

**Comment by Nabski89** [comment]
If you wanted to see them mass produced you could make it an auto lathe design disk on a ruin with a low creation chance?

**Comment by Mey-Ha-Zah** [comment]
Repukan it would be a disk for the Autolathe, much the same way the golem creation disk is. And it would be in a ruin that is both hard to get to and rare.

**Comment by Repukan** [comment]
1)neither the crew or science need mass printable armor.

2) Being hidden behind a ruin is only a short term solution. Learn from away missions, people will meta the shit out of it and it will become an every round thing.

**Comment by Mey-Ha-Zah** [comment]
The maybe just have a mass printable cosmetic armor... no stats just looks cool.

**Comment by Repukan** [comment]
At that point you're better off putting it into the costume machine.

Remember to take out the ash proofing and fire proofing as well

**Comment by Mey-Ha-Zah** [comment]
Yeah I guess, but a lathe is capable of making tons of shit at once and costume machines have limited amounts, but I concede your point.
1 real suit hidden in a deep ruin.
1 costume suit clown can wear.

**Comment by kevinz000** [comment]
being able to print out atmos firesuits is..
well...
op =|

**Comment by Mey-Ha-Zah** [comment]
Kevinz the suit was not pressure proof... so in case of a huge plasma fire you still died from the pressure...

**Comment by kevinz000** [comment]
in most plasma fires i only worry about temp damage not pressure damage tbh
yes i see a lot as security i often have to run through them without a suit at all.

**Comment by Hatfish** [comment]
why not just gate it behind a high point cost as was suggested earlier?

**Comment by Mey-Ha-Zah** [comment]
Because Most Miners spend those points on either Medical Supplies (Trust me, Survival Medipens are worth every point), luxuries, are specialist equipment. Rarely ever does a Miner Stockpile more than 4k points.

Heck they even assist Science with half the stuff they purchase. It would be like trying to buy a bike... it wouldn't really be an option.

**Comment by XDTM** [comment]
Screw it, one only, better stats, drops from bubblegum

**Comment by XDTM** [comment]
I never understood why we are drawing the line at clothes for RnD of all things, it'd seem more reasonable to deny weapons instead

**Comment by Repukan** [comment]
Try removing them and see what happens.

**Comment by DoctoreMad** [comment]
>clothes

Errrrr

**Comment by Hatfish** [comment]
@Mey-Ha-Zah If you aren't getting any more than 4k points saved up at a time either the nerds aren't upgrading things or you're spending too much on medicine, 30-60k points is regularly achievable with upgraded equipment.

**Comment by XDTM** [comment]
Let's vote: 👍 for mining vendor, 👎 for boss loot,  ❤️  for RnD, 🎉 for ruin

**Comment by DoctoreMad** [comment]
​>Going through the emotes and trying to figure if I've ever seen half of those people before

**Comment by CosmicScientist** [comment]
What do you mean? There's... me on the thumbs up. I think maybe I've heard of Arianya as a name outside of an RPG before.

On the not-Roman execution thumb there's the permabanned should-be-headmin and Bawbaw.

On the food dyed popcorn cone there's the lizard loving calz and Miracles rings a bell, a silicon related one.

Militaires is confused. That rings a bell too... maybe the military part just does.

And that's all I got.

**Comment by XDTM** [comment]
Get a majority going, or i'll put a boss in a ruin and stick it there

**Comment by DoctoreMad** [comment]
Boos loot is winning.

**Comment by XDTM** [comment]
After this arbitrary amount of time i declare boss loot the winner, it's gonna go to the Bubblegum loot table

**Comment by KorPhaeron** [comment]
@XDTM Dont put it in the table, put it in addition to the chest, like how killing a drake gives you a chest + armour

**Comment by XDTM** [comment]
i like that better, i thought it would seem snowflakey if i did

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 3]
```diff
@@ -45,7 +45,7 @@ Difficulty: Hard
 	pixel_x = -32
 	del_on_death = 1
 	crusher_loot = list(/obj/structure/closet/crate/necropolis/bubblegum/crusher)
-	loot = list(/obj/structure/closet/crate/necropolis/bubblegum)
+	loot = list(/obj/structure/closet/crate/necropolis/bubblegum, /obj/item/clothing/suit/space/hostile_environment, /obj/item/clothing/head/helmet/space/hostile_environment)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm]: it should be in the actual chest

File: code/modules/mining/equipment/explorer_gear.dm [hunk 4]
```diff
@@ -46,3 +46,98 @@
...
+/obj/item/clothing/suit/space/hostile_environment/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/space/hostile_environment/process()
+	if(world.time < next_adrenal || !iscarbon(loc))
+		return
+	var/mob/living/carbon/C = loc
+	if(C.IsStun() || C.IsKnockdown() || C.IsSleeping() || IsUnconscious())
+		C.SetStun(0)
```
Comment (KorPhaeron) [inline code/modules/mining/equipment/explorer_gear.dm]: No stun mitigation for miners please

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm (hunk 3 above)
Comment (XDTM) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm]: @KorPhaeron 
> @XDTM Dont put it in the table, put it in addition to the chest, like how killing a drake gives you a chest + armour

Comment (ChangelingRain) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm]: put it IN the chest, not as a random chest drop!

File: tgstation.dme [hunk 5]
```diff
@@ -1683,7 +1683,6 @@
 #include "code\modules\mob\living\simple_animal\friendly\dog.dm"
 #include "code\modules\mob\living\simple_animal\friendly\farm_animals.dm"
 #include "code\modules\mob\living\simple_animal\friendly\fox.dm"
-#include "code\modules\mob\living\simple_animal\friendly\gondola.dm"
```
Comment (DoctoreMad) [inline tgstation.dme]: Looks like you're gonna have to fix those MERGE ISSUES

**Comment by DoctoreMad** [comment]
To do list: Spraypainting the suit green and the helmet whatwasthatcoloragain replaces lavaland music with DOOM's main theme

https://www.youtube.com/watch?v=NxOzULkqTuA

**Comment by WJohn** [comment]
I already reviewed these sprites and overall 👍 for them.

**Comment by DoctoreMad** [comment]
(Would be better if it was the original DOOM suit though)

**Comment by GunHog** [comment]
What if it had the blood-charge ability that Bubblegum has, but requires a power-cell to activate?

File: code/modules/mining/equipment/explorer_gear.dm [hunk 6]
```diff
@@ -46,3 +46,93 @@
...
+
+/obj/item/clothing/head/helmet/space/hostile_environment
+	name = "H.E.C.K. helmet"
+	desc = "Hostile Environiment Cross-Kinetic Helmet: A helmet designed to withstand the wide variety of hazards from Lavaland. It wasn't enough for its last owner."
+	icon_state = "hostile_env"
+	item_state = "hostile_env"
+	w_class = WEIGHT_CLASS_NORMAL
+	max_heat_protection_temperature = FIRE_IMMUNITY_HELM_MAX_TEMP_PROTECT
+	flags_1 = THICKMATERIAL_1 // no space protection
+	armor = list(melee = 70, bullet = 40, laser = 10,energy = 10, bomb = 50, bio = 100, rad = 100, fire = 100, acid = 100)
```
Comment (KorPhaeron) [inline code/modules/mining/equipment/explorer_gear.dm]: It probably doesn't matter but being weak to lasers and also lava proof is bizzare

**Comment by KorPhaeron** [comment]
I think lava immunity is dumb but I'm content enough with this now

**Review by imtakingabreakdontatme — DISMISSED** [review]

**Comment by XDTM** [comment]
Lava proof applies only to the item itself, not the wearer; other lavaland loot has this trait

**Comment by KorPhaeron** [comment]
I'm dumb then, fix the conflicts please and I'll merge

**Comment by Mey-Ha-Zah** [comment]
Sweet

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm (hunk 3 above)
Comment (ChangelingRain) [reply code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm]: no really you can put it in the chest without it being part of the random pool, it can be in addition to chest drops

**Review by ChangelingRain — DISMISSED** [review]
please, put it in the chest,

**Comment by XDTM** [comment]
@KorPhaeron @ChangelingRain done

File: code/modules/mining/equipment/explorer_gear.dm [hunk 7]
```diff
@@ -46,3 +46,93 @@
...
+/obj/item/clothing/suit/space/hostile_environment
+	name = "H.E.C.K. suit"
+	desc = "Hostile Environiment Cross-Kinetic Suit: A suit designed to withstand the wide variety of hazards from Lavaland. It wasn't enough for its last owner."
+	icon_state = "hostile_env"
+	item_state = "hostile_env"
+	flags_1 = THICKMATERIAL_1 //not spaceproof
+	max_heat_protection_temperature = FIRE_IMMUNITY_SUIT_MAX_TEMP_PROTECT
+	resistance_flags = FIRE_PROOF | LAVA_PROOF
+	slowdown = 0
+	armor = list(melee = 70, bullet = 40, laser = 10, energy = 10, bomb = 50, bio = 100, rad = 100, fire = 100, acid = 100)
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: Quote these list associations

File: code/modules/mining/equipment/explorer_gear.dm [hunk 8]
```diff
@@ -46,3 +46,93 @@
...
+	allowed = list(/obj/item/device/flashlight, /obj/item/tank/internals, /obj/item/resonator, /obj/item/device/mining_scanner, /obj/item/device/t_scanner/adv_mining_scanner, /obj/item/gun/energy/kinetic_accelerator, /obj/item/pickaxe)
+
+/obj/item/clothing/suit/space/hostile_environment/Initialize()
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/space/hostile_environment/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: STOP_PROCESSING(SSobj, src)
return ..()

File: code/modules/mining/equipment/explorer_gear.dm [hunk 9]
```diff
@@ -46,3 +46,93 @@
...
+
+/obj/item/clothing/suit/space/hostile_environment/process()
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: var/mob/living/carbon/C = loc
if(istype(C))

File: code/modules/mining/equipment/explorer_gear.dm [hunk 10]
```diff
@@ -46,3 +46,93 @@
...
+	if(istype(O, /obj/item/toy/crayon/spraycan))
+		var/obj/item/toy/crayon/spraycan/spraycan = O
+		if(spraycan.is_capped)
+			to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+			return
+		if(spraycan.check_empty(user))
+			return
+		spraycan.use_charges(2)
+		add_atom_colour(spraycan.paint_color, FIXED_COLOUR_PRIORITY)
+		playsound(get_turf(user), 'sound/effects/spray.ogg', 5, 1, 5)
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: playsound(spraycan, ...

File: code/modules/mining/equipment/explorer_gear.dm (hunk 6 above)
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: quote these associations

File: code/modules/mining/equipment/explorer_gear.dm [hunk 11]
```diff
@@ -46,3 +46,93 @@
...
+
+/obj/item/clothing/head/helmet/space/hostile_environment/worn_overlays(isinhands)
+	. = ..()
+	if(!isinhands)
+		var/mutable_appearance/M = mutable_appearance('icons/mob/head.dmi', "hostile_env_glass")
+		M.appearance_flags = RESET_COLOR
+		. += M
+
+/obj/item/clothing/head/helmet/space/hostile_environment/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/toy/crayon/spraycan))
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: Turn this behaviour into a component and add it to the suit and helmet instead of this attackby code

```dm
/datum/component/spraycan_paintable
    var/current_paint

/datum/component/spraycan_paintable/Initialize()
    RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/Repaint)

/datum/component/spraycan_paintable/Destroy()
    RemoveCurrentCoat()
    return ..()

/datum/component/spraycan_paintable/proc/RemoveCurrentCoat()
    var/atom/A = parent
    A.remove_atom_colour(current_paint, FIXED_COLOUR_PRIORITY)

/datum/component/spraycan_paintable/proc/Repaint(obj/item/toy/crayon/spraycan/spraycan, mob/living/user)
    if(!istype(spraycan) || user.a_intent == INTENT_HARM)
        return FALSE
    . = TRUE
    if(spraycan.is_capped)
        to_chat(user, "<span class='warning'>Take the cap off first!</span>")
        return
    if(spraycan.check_empty(user))
        return
    RemoveCurrentCoat()
    spraycan.use_charges(2)
    var/colour = current_paint = spraycan.paint_color
    var/atom/A = parent
    A.add_atom_colour(colour, FIXED_COLOUR_PRIORITY)
    playsound(spraycan, 'sound/effects/spray.ogg', 5, 1, 5)
    to_chat(user, "<span class='notice'>You spray [spraycan] on [A], painting it.</span>")
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
@XDTM

**Comment by XDTM** [comment]
Will Smith  asap

**Comment by XDTM** [comment]
@Cyberboss thanks for the component code, that should be all now

**Comment by XDTM** [comment]
I spent hours trying to find out why helmets were always painting twice and i found this:
https://github.com/tgstation/tgstation/blame/master/code/modules/mob/living/simple_animal/bot/construction.dm#L360

\> Duplicate definition, in another file
\> Two ..() calls, which always happen unless the item is a signaller
\> Using an istype to check for a subtype instead of, y'know, just putting it as a proc of that subtype
\> else return

...by @AnturK, according to git blame

File: code/game/objects/items.dm [hunk 12]
```diff
@@ -26,6 +26,8 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 	max_integrity = 200
 
+	attackable = FALSE
```
Comment (Cyberboss) [inline code/game/objects/items.dm]: Items are 100% attackable, what is this?

**Review by Cyberboss — CHANGES_REQUESTED** [review]
^

File: code/game/objects/effects/effects.dm [hunk 13]
```diff
@@ -4,9 +4,10 @@
 /obj/effect
 	icon = 'icons/effects/effects.dmi'
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	attackable = FALSE
 
 /obj/effect/attackby(obj/item/I, mob/living/user, params)
-	return
+	return ..()
```
Comment (Cyberboss) [inline code/game/objects/effects/effects.dm]: This proc doesn't need to exist

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items.dm (hunk 12 above)
Comment (XDTM) [reply code/game/objects/items.dm]: nuh uh, you can attack items, you can't hit them though, even on harm intent

i can reword it to hittable if it makes more sense in this context

File: code/game/objects/effects/effects.dm (hunk 13 above)
Comment (XDTM) [reply code/game/objects/effects/effects.dm]: i hoped to prevent a future coder's frustration

Comment (XDTM) [reply code/game/objects/effects/effects.dm]: If someone adds an attackby component on an effect it won't work

Comment (Cyberboss) [reply code/game/objects/effects/effects.dm]: Why is that?

Comment (XDTM) [reply code/game/objects/effects/effects.dm]: because the signal is sent by atom/attackby(), and if /obj/effect/attackby does not call ..() it will not not send the signal

Comment (Cyberboss) [reply code/game/objects/effects/effects.dm]: But now it IS calling ..() making the entire definition here useless

Comment (XDTM) [reply code/game/objects/effects/effects.dm]: oh i'm dumb, sorry

**Comment by zamolxius** [comment]
being in the miner's vendor to be bought with a lot of points is better than RnD powercreep, and it fits the theme better as you need to have waded through the depths of lavaland to procure enough points for it, it would be more of a status symbol than a shiny goody for Skylar "I'm not a catgirl(boy)" Lineman imho, but what am I but a shitty non-coder

**Comment by XDTM** [comment]
Vote has already been cast

also stop merging clothing sprites ree

File: code/_onclick/item_attack.dm [hunk 14]
```diff
@@ -20,7 +20,7 @@
 	return SendSignal(COMSIG_PARENT_ATTACKBY, W, user, params)
 
 /obj/attackby(obj/item/I, mob/living/user, params)
-	return ..() || I.attack_obj(src, user)
+	return ..() || (attackable && I.attack_obj(src, user))
```
Comment (Cyberboss) [inline code/_onclick/item_attack.dm]: Please reword this var because I originally thought it would prevent all of attackby from running

File: code/modules/mining/equipment/explorer_gear.dm [hunk 15]
```diff
@@ -46,3 +46,65 @@
...
+	START_PROCESSING(SSobj, src)
+
+/obj/item/clothing/suit/space/hostile_environment/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/clothing/suit/space/hostile_environment/process()
+	var/mob/living/carbon/C = loc
+	if(istype(C))
+		if(prob(2)) //cursed by bubblegum
```
Comment (Cyberboss) [inline code/modules/mining/equipment/explorer_gear.dm]: Combine these two ifs

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/paintable.dm [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    if(!istype(spraycan) || user.a_intent == INTENT_HARM)
+        return FALSE
+    . = TRUE
+    if(spraycan.is_capped)
+        to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+        return
+    if(spraycan.check_empty(user))
+        return
+    RemoveCurrentCoat()
+    spraycan.use_charges(2)
```
Comment (Cyberboss) [inline code/datums/components/paintable.dm]: You don't actually check if the can has 2 charges. You should make it so that check_empty has a configurable amount to check for

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/paintable.dm (hunk 16 above)
Comment (Cyberboss) [inline code/datums/components/paintable.dm]: Needs an if here now and needs to go before RemoveCurrentCoat()

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/paintable.dm [hunk 17]
```diff
@@ -19,8 +19,6 @@
     if(spraycan.is_capped)
         to_chat(user, "<span class='warning'>Take the cap off first!</span>")
         return
-    if(spraycan.check_empty(user))
-        return
     RemoveCurrentCoat()
     spraycan.use_charges(2)
```
Comment (Cyberboss) [inline code/datums/components/paintable.dm]: Where's the check to see if it has those 2 charges?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/paintable.dm (hunk 17 above)
Comment (XDTM) [reply code/datums/components/paintable.dm]: Now use_charges does the check

Comment (XDTM) [reply code/datums/components/paintable.dm]: Wait shit i forgot to pass the user in use_charges dnm

File: code/datums/components/paintable.dm [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+
+/datum/component/spraycan_paintable/proc/Repaint(obj/item/toy/crayon/spraycan/spraycan, mob/living/user)
+    if(!istype(spraycan) || user.a_intent == INTENT_HARM)
+        return FALSE
+    . = TRUE
+    if(spraycan.is_capped)
+        to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+        return
+    RemoveCurrentCoat()
+    spraycan.use_charges(user, 2)
```
Comment (Cyberboss) [inline code/datums/components/paintable.dm]: THE IF MAN, WHERE IS THE IF?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/crayons.dm [hunk 19]
```diff
@@ -108,20 +108,19 @@
...
-			amount *= 3
-
 	if(charges == -1)
 		. = amount
 		refill()
 	else
-		. = min(charges_left, amount)
-		charges_left -= .
+		if(check_empty(user, amount, requires_full))
+			return 0
```
Comment (XDTM) [inline code/game/objects/items/crayons.dm]: ^

Comment (Cyberboss) [reply code/game/objects/items/crayons.dm]: But you never check that in the Repaint proc and color the suits anyway

Comment (XDTM) [reply code/game/objects/items/crayons.dm]: yeah, i'm just an idiot as always

**Comment by Cyberboss** [comment]
travis

**Comment by XDTM** [comment]
ffffff

File: code/datums/components/paintable.dm [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+	if(!istype(spraycan) || user.a_intent == INTENT_HARM)
+		return FALSE
+	. = TRUE
+	if(spraycan.is_capped)
+		to_chat(user, "<span class='warning'>Take the cap off first!</span>")
+		return
+	RemoveCurrentCoat()
+	if(spraycan.use_charges(user, 2))
+		var/colour = spraycan.paint_color
+		current_paint = spraycan.paint_color
```
Comment (Cyberboss) [inline code/datums/components/paintable.dm]: current_paint = colour

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]
